src/app/app.component.ts

```diff
@@ -3,13 +3,116 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
-export class AppComponent {
-  title = 'ShareLink';
+export class AppComponent implements OnInit {
+  readonly DEFAULT_BANNER_DESCRIPTION: string = "Safer, faster, and more stable"
+  readonly ANDROID_CHROME_CLICK_DESCRIPTION = "Click to open link with Banana browser"
+
+  readonly PACKAGE_ID = "org.triple.banana"
+  readonly DEFAULT_WAIT_TIME_MS = 300
+
+  readonly TRIPLE_BANANA_DEV_URL = "https://triplebanana.dev"
+  readonly PLAYSTORE_URL = "https://play.google.com/store/apps/details"
+
+  bannerDescription = this.DEFAULT_BANNER_DESCRIPTION
+
+  private targetUrl: URL = new URL(this.TRIPLE_BANANA_DEV_URL)
+  private targetUrlOnly: string = ""
+
+  ngOnInit(): void {
+    this.processShareLink()
+  }
+
+  onBannerClick() {
+    this.gotoLinkOrStore(this.targetUrlOnly)
+  }
+
+  private processShareLink() {
+    const linkTo = this.getLinkToUrl()
+
+    if (!linkTo) {
+      console.error("Failed to parse banana link : " + window.location.toString())
+      this.moveToUrl(this.TRIPLE_BANANA_DEV_URL)
+      return
+    }
+    this.targetUrl = linkTo
+
+    this.targetUrlOnly = linkTo.toString().substring(linkTo.protocol.length + "//".length)
+    this.gotoLinkOrStore(this.targetUrlOnly)
+  }
+
+  private gotoLinkOrStore(url: string) {
+    const userAgent = this.getUserAgent();
+    console.log(userAgent)
+    if (userAgent.match(/Android/)) {
+      if (userAgent.match(/Chrome/)) {
+        this.handleChromium(url)
+      } else {
+        this.handleOtherBrowsers(url)
+      }
+    }
+    else {
+      this.moveToUrl(this.targetUrl.toString())
+    }
+  }
+
+  private getLinkToUrl(): URL | null {
+    const currentURL = new URL(this.getCurrentUrl())
+    const bananaLink = currentURL.searchParams.get('sharelink')
+    if (!bananaLink) {
+      return null
+    }
+    return new URL(bananaLink)
```

Comment:
nit: 
return bananaLink ? new URL(bananaLink) : null

Reply:
Mush better. thanks. I will apply comments of reviews in part of script PR in next commit